recipes/NavierStokes/meta.yaml

```
@@ -0,0 +1,50 @@
+{% set name = "iscd-nstokes" %}
+{% set version = "0.0.0rc0" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  # there are no tagged releases which so using archives of specific commits
+  url: https://github.com/ISCDtoolbox/NavierStokes/archive/2e14ef5fa4884674b7bbc2212af256652f02912c.zip
+  sha256: 7efe022f2b3ff88001b99c9cb83da5bdd15fb2b2084201662a55e4d666157116
+
+build:
+  number: 0
+  skip: true  # [py<3 or osx or win] # Cannot open include file: 'pthread.h': No such file or directory  
+  ignore_run_exports_from:
+    - {{ compiler('cxx') }}
```

```suggestion
build:
  number: 0
  skip: true  # [py<3 or osx or win] # Cannot open include file: 'pthread.h': No such file or directory  
  ignore_run_exports_from:
    - {{ compiler('cxx') }}
  run_exports:
    - {{ pin_subpackage('iscd-nstokes', exact=True) }}
```
For the packages which ship libraries and headers (which is all?), you need to include run_exports of the same package. This ensures that downstream packages always depend on the same version that was used to build them. For your packages which do not have official releases, use `exact=True` and for the official releases use `max_pin='x.x.x'`.

---

Hi, I added run_exports to every packages as you suggested. I tried very hard (a few hours of tests) to use `exact=True` but I couldn't : the package could not be found when later I used it as a dependency in `mmg`. The issue seems similar to this one : https://github.com/conda/conda-build/issues/4415 but no workaround seemed to work in my case. It looks like the build will look for a `0` build package number when it should use the hash instead. The only solution I could find was to used the `max_pin='x.x.x'` for all cases. I think (I might be wrong) that the risk of regression is limited. If you have other tests that you wish me to do please let me know and I will do my best :-)

---

Thanks for trying to debug that issue.

In this case, we will use `{{ version }}` as the pinning. This will pin the version, but not the build number. Otherwise, by using `max_pin='x.x.x'` we get a `>=0.0.0rc0,<0.1.0` constraint which is not correct because it would allow `0.0.0rc1` at runtime.